src/main/java/de/tum/in/www1/artemis/web/rest/StudentExamResource.java

```diff
@@ -334,6 +340,57 @@ private StudentExam findStudentExamWithExercisesElseThrow(User user, Long examId
         return studentExam;
     }
 
+    /**
+     * GET /courses/{courseId}/exams/{examId}/student-exams/grade-summary : Return student exam result, aggregate points, assessment result
+     * for a student exam and grade calculations if the exam is assessed. Only instructors can use userId parameter to get exam results of other users,
+     * if the caller is a student, userId should either be the user id of the caller or empty.
+     *
+     * Does not return the student exam itself to save bandwidth.
+     *
+     * See {@link StudentExamWithGradeDTO} for more explanation.
+     *
+     * @param courseId  the course to which the student exam belongs to
+     * @param examId    the exam to which the student exam belongs to
+     * @param userId    the user id of the student whose grade summary is requested
+     * @return the ResponseEntity with status 200 (OK) and with the StudentExamWithGradeDTO instance without the student exam as body
+     */
+    @GetMapping("/courses/{courseId}/exams/{examId}/student-exams/grade-summary")
+    @PreAuthorize("hasRole('USER')")
+    public ResponseEntity<StudentExamWithGradeDTO> getStudentExamGradesForSummary(@PathVariable Long courseId, @PathVariable Long examId,
+            @RequestParam(required = false) Long userId) {
+        long start = System.currentTimeMillis();
+        User currentUser = userRepository.getUserWithGroupsAndAuthorities();
+        log.debug("REST request to get the student exam grades of user with id {} for exam {} by user {}", userId, examId, currentUser.getLogin());
+
+        User targetUser = userId == null ? currentUser : userRepository.findByIdWithGroupsAndAuthoritiesElseThrow(userId);
+        StudentExam studentExam = findStudentExamWithExercisesElseThrow(targetUser, examId, courseId);
+
+        boolean isAtLeastInstructor = authorizationCheckService.isAtLeastInstructorInCourse(studentExam.getExam().getCourse(), currentUser);
+        if (!isAtLeastInstructor && !currentUser.getId().equals(targetUser.getId())) {
+            throw new AccessForbiddenException("Current user cannot access grade info for target user");
+        }
+
+        // check that the studentExam has been submitted, otherwise /student-exams/conduction should be used
+        if (!studentExam.isSubmitted() || !studentExam.areResultsPublishedYet()) {
+            throw new AccessForbiddenException();
```

Comment:
I would expect an error message :thinking:

Reply:
That condition is analogous to the condition at [`/courses/{courseId}/exams/{examId}/student-exams/summary` endpoint](https://github.com/ls1intum/Artemis/pull/4932/files/30a634968d82a3235bd75186945580bc56ce2f8b#diff-8098ab9f22aba5ddb5a01571b4841f329ac30ab0ff3bce8d32223f8b1c991b6fL317) in the [StudentExamResource.java](https://github.com/ls1intum/Artemis/pull/4932/files/30a634968d82a3235bd75186945580bc56ce2f8b#diff-8098ab9f22aba5ddb5a01571b4841f329ac30ab0ff3bce8d32223f8b1c991b6f) file (you may need to expand the diff to see line 323 because it is hidden in the unchanged part. So I kept it the same with the other endpoint (without message).